environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/base/EnvironmentNetworkBase.java

```diff
@@ -42,6 +42,12 @@
     @ApiModelProperty(EnvironmentModelDescription.SERVICE_ENDPOINT_CREATION)
     private ServiceEndpointCreation serviceEndpointCreation = ServiceEndpointCreation.DISABLED;
 
+    @ApiModelProperty(EnvironmentModelDescription.PUBLIC_ENDPOINT_ACCESS_GATEWAY)
+    private Boolean usePublicEndpointAccessGateway = false;
```

Comment:
Usually, we prefer enums instead of booleans, it makes it easier to extend it the future.